wpilib/wpilib/smartdashboard.py

```diff
@@ -380,12 +378,11 @@ def setDefaultBooleanArray(cls, key, defaultValue):
         return table.setDefaultBooleanArray(key, defaultValue)
 
     @classmethod
-    def getBooleanArray(cls, key, defaultValue):
+    def getBooleanArray(cls, key: str, defaultValue: List[bool]) -> List[bool]:
```

Comment:
NetworkTables will currently never return a list. Better to specify `Sequence[bool]`.

Reply:
I'm gonna use Iterable because networktables has type hints for iterable in some places

Reply:
Last I checked, NetworkTables doesn't even have type hints?

Reply:
I think it would be better to specify Tuple explicitly, because it will always return a Tuple as that's immutable.

Reply:
The input default value could be anything -- but the return value will always be a tuple, and I'm perfectly happy with making that API.

Reply:
the return won't be a Tuple if it returns the default value though

Reply:
typing.Tuple[] would be incorrect - a Tuple has a fixed length. A Sequence[T] is an arbitrary length sequence (i.e. implements `__len__` and `__getitem__`) of T, which is correct.

Reply:
If we want to denote that the default value can be any type, it would be appropriate to have a type variable T, which is the type of the default value, and return a Union[T, Sequence[...]].